packages/core/src/crypto/jose/jwk/P_256Jwk.ts

```diff
@@ -0,0 +1,102 @@
+import type { JwkJson } from './Jwk'
+
+import { TypedArrayEncoder, Buffer } from '../../../utils'
+import { KeyType } from '../../KeyType'
+import { JwaCurve, JwaKeyType } from '../jwa'
+import { JwaSignatureAlgorithm } from '../jwa/alg'
+
+import { Jwk } from './Jwk'
+import { compress, expand } from './ecCompression'
+import { hasKty, hasCrv, hasX, hasY, hasValidUse } from './validate'
+
+export class P_256Jwk extends Jwk {
```

Comment:
Sure!

Reply:
Would you also want to rename the keys for all these algorithsm then?

```
export enum JwaEncryptionAlgorithm {
  RSA1_5 = 'RSA1_5',
  RSA_OAEP = 'RSA-OAEP',
  RSA_OAEP_256 = 'RSA-OAEP-256',
  A128KW = 'A128KW',
  A192KW = 'A192KW',
  A256KW = 'A256KW',
  dir = 'dir',
  ECDH_ES = 'ECDH-ES',
  ECDH_ES_A128KW = 'ECDH-ES+A128KW',
  ECDH_ES_A192KW = 'ECDH-ES+A192KW',
  ECDH_ES_A256KW = 'ECDH-ES+A256KW',
  A128GCMKW = 'A128GCMKW',
  A192GCMKW = 'A192GCMKW',
  A256GCMKW = 'A256GCMKW',
  PBES2_HS256_A128KW = 'PBES2-HS256+A128KW',
  PBES2_HS384_A192KW = 'PBES2-HS384+A192KW',
  PBES2_HS512_A256KW = 'PBES2-HS512+A256KW',
}
```

Becaues I've now changed the JwaCurve to also remove `_` to make it consistent, but then we should do it everywhere (consistency is important here I think, because otherwise we're going to have inconsistencies down the line)
```
export enum JwaCurve {
  P256 = 'P-256',
  P384 = 'P-384',
  P521 = 'P-521',
  Ed25519 = 'Ed25519',
  X25519 = 'X25519',
}
```

Reply:
Yeah I think renaming the algorithms is also fine just to stay inline with TypeScript enum naming conventions.